Documentation/specs/corehost.md

```diff
@@ -0,0 +1,112 @@
+
+Given the set of assembly names, the host performs the following resolution. In some steps, the Package ID/Version/Relative Path data is required. This is only available if the assembly was listed in the deps file. If the assembly comes from the app-local folder, these resolution steps are skipped.
+
+1. If there is an entry in the servicing index for the Package ID/Version/Relative Path associated with the assembly, the Servicing Root is concatenated with the New Asset Path from the index and used as the Assembly Path. This occurs **even if the assembly is also located app-local**, as long as it is also in the `.deps` file.
+2. If there is a file in the Application Base with the file name `[AssemblyName].dll`, `[AssemblyName].ni.dll`, `[AssemblyName].exe`, or `[AssemblyName].ni.exe` (in that order), it its full path is used as the Assembly Path
+3. The Assembly Path is resolved out of the Package Caches using the algorithm above (in 'Files from package caches').
+
+A similar process is used to produce a list of Native Library Paths. Native libraries are listed in the `.deps` file (marked with a special `native` type token) and searched in the same way as managed assemblies (Servicing, then app-local, then package caches). The main exception is that the app-local file extensions vary by platform (`.dll` on Windows, `.so` on Linux, `.dylib` on Mac OS X)
+
+Once a the list of assemblies and native libraries is produced, each is deduplicated by taking the first match for an assembly/file name (**open question:** error if two assemblies with the same name?). The managed assemblies are provided in the Trusted Platform Assemblies (TPA) list for the CoreCLR during startup. The folder paths for each native library are deduplicated and provided in the Native Search Paths list for the CoreCLR during startup.
+
+**NOTE**: The CLR may add support for providing a similar structure as the TPA list for native libraries (i.e. a flat list of full file paths).
+
+### Satellite Assemblies
+
+Satellite Assemblies (assemblies containing only embedded resources used in place of the resources provided by an assembly) are stored just like regular assemblies in the TPA list. The CLR will use a path convention to detect which assemblies are culture-neutral and which are culture-specific. The convention will be to look at the last two segments of the path (the file name and the immediate parent directory name). If the parent directory matches an [IETF Language Tag](https://en.wikipedia.org/wiki/IETF_language_tag) (or more specifically, a value usable in the Culture field for a CLR Assembly Name), then the assembly is considered culture-specific (for the culture specified in that folder name).
+
+**NOTE**: The CLR does not do this today, but there will be a work item to track it.
+
+## Runtime Resolution
+
+Runtime resolution is controlled by two environment variables:
+
+* `DOTNET_RUNTIME_SERVICING` -> Global override for runtime
```

Comment:
This was what we discussed, but you're right it could probably be in the servicing index (right now we don't want to put it in the deps file because we don't want to couple the app to the runtime unless the runtime is published app-local). The immediate concern that comes to mind is that right now we don't identify any runtime version data.

This is generally an open discussion so I will add some comments to that effect to this spec. We can continue that discussion after the spec is merged (to avoid blocking other decisions). Nothing else here should block this discussion.